### PR TITLE
[RFC][NI] Add (limited) tree-shaking capabilities

### PR DESCRIPTION
**Pull request by AndreJoaquim**
Closes https://github.com/uniplaces/ember-cli-uniq/issues/279.

### CODE REVIEW

File: addon/styles/addon.scss [hunk 1]
```diff
@@ -1,46 +1,46 @@
 @import "node_modules/uniplaces-uniq/src/styles/_main";
 
-@import "partials/uni-textarea",
+@import "partials/uni-alert",
```
Comment (AndreJoaquim) [inline addon/styles/addon.scss:3]: Ordered by natural order

**Review by jkrnak — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -1,5 +1,100 @@
...
+    let app = this._findHost();
+
+    this.app = app;
+
+    this.emberCliUniqOptions = app.options[this.name] || {};
+    this.whitelist = this.generateWhitelist(this.emberCliUniqOptions.only);
+  },
+
+  treeForAddon() {
+    var tree = this._super.treeForAddon.apply(this, arguments);
```
Comment (rntdrts) [inline index.js]: change the usage of var to let or const whatever you see more fit...

File: index.js [hunk 3]
```diff
@@ -1,5 +1,100 @@
...
+  'uni-textarea': ['uni-tooltip'],
+  'uni-title': ['uni-subtitle']
+}
+
 module.exports = {
-  name: 'ember-cli-uniq'
+  name: require('./package').name,
+
+  included() {
+    this._super.included.apply(this, arguments);
```
Comment (rntdrts) [inline index.js:29]: no easy way to do this?
and out of curiosity what would be the super of this?

File: index.js [hunk 4]
```diff
@@ -1,5 +1,100 @@
...
+
+    let app = this._findHost();
+
+    this.app = app;
```
Comment (rntdrts) [inline index.js]:
```suggestion
    this.app = this._findHost();
    this.emberCliUniqOptions = Object.assign({}, this.app.options[this.name]);
```

File: index.js (hunk 3 above)
Comment (AndreJoaquim) [reply index.js:29]: I don't think so, at least all addons I've checked do it that way, e.g. https://github.com/DockYard/ember-composable-helpers/blob/master/index.js#L12

File: index.js (hunk 4 above)
Comment (rntdrts) [reply index.js]: unless you want to have some kind of shared scope changes with that option and the `this.emberCliUniqOptions`...

File: index.js (hunk 3 above)
Comment (AndreJoaquim) [reply index.js:29]: I've checked the API documentation and it has the same syntax: https://ember-cli.com/api/classes/Addon.html (search for `public included(parent)`)

**Review by rntdrts — APPROVED** [review]
I just left some suggestions and some doubts... nothing major though.

well done! 👏

File: index.js (hunk 3 above)
Comment (rntdrts) [reply index.js:29]: Thanks, @AndreJoaquim 🙏 !!!

**Pull request merged**